src/monomorphize.rs

```diff
@@ -11,3 +11,12 @@ pub fn apply_param_substs<'a, 'tcx,T>(tcx: &TyCtxt<'a, 'tcx, 'tcx>,
     let substituted = value.subst(*tcx, param_substs);
     tcx.normalize_associated_type(&substituted)
 }
+
+pub fn apply_ty_substs<'a, 'tcx>(tcx: &TyCtxt<'a, 'tcx, 'tcx>,
```

Comment:
Is it possible to share this with rustc? It seems unfortunate to have to reimplement monomorphization.